src/find/matchers/fs.rs

```diff
@@ -24,14 +31,24 @@ use super::Matcher;
 ///
 /// This is only supported on Unix.
 #[cfg(unix)]
-pub fn get_file_system_type(path: &Path) -> Result<String, Box<dyn Error>> {
+pub fn get_file_system_type(
+    path: &Path,
+    cache: &RefCell<Option<Cache>>,
+) -> Result<String, Box<dyn Error>> {
     use std::os::unix::fs::MetadataExt;
 
-    let metadata = match path.metadata() {
+    let metadata = match path.symlink_metadata() {
```

Comment:
Could you describe why `symlink_metadata` is used here instead of `metadata`? (If possible please mark it with a comment in the code). thanks.

Reply:
`symlink_metadata` corresponds to `lstat` which works like `stat` on regular files/dirs but does not follow a symlink. This is aligned with GNU find because it uses `AT_SYMLINK_NOFOLLOW` flag. I added comments for this and also a test in `find_fs` to capture this.